### PR TITLE
MINIFI-39 Adding PeriodicStatusReporter to the bootstrap process with…

### PR DESCRIPTION
**Pull request by JPercivall**
… an implementation to log the FlowStatus

### CODE REVIEW

**Comment by thesolson** [comment]
Hi Joe,
Had a quick look at the Admin Guide and it looks great! Am suggesting some minor changes here: https://github.com/thesolson/nifi-minifi/commit/6d0cf996edbea7f3239d4819b202facb01243c5d

**Comment by apiri** [comment]
reviewing

File: minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/status/PeriodicStatusReporter.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public abstract void initialize(Properties properties, QueryableStatusAggregator queryableStatusAggregator);
+
+    /**
+     * Begins the associated reporting service provided by the given implementation.  In most implementations, no action will occur until this method is invoked. The implementing class must have set
+     * 'reportRunner' prior to this method being called.
+     *
+     */
+    public void start() {
+        if (reportRunner == null){
+            throw new IllegalStateException("Programmatic error, the reportRunner has is still NULL when 'start' was called.");
```
Comment (apiri) [inline minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/status/PeriodicStatusReporter.java]: minor:  has is -> is

**Comment by JPercivall** [comment]
@apiri just added @thesolson doc changes (thanks Sarah!) and added another commit of my own which addresses the "has is" -> "is" and rewords the first paragraph of the system admin guide a bit.

**Comment by apiri** [comment]
@JPercivall ah, good catches.  All looks good here.

**Comment by JPercivall** [comment]
Rgr, I'll go ahead and merge it. Thanks @thesolson and @apiri!

**Pull request closed without merging**